### PR TITLE
KVS extensions

### PR DESCRIPTION
**Pull request by adecaro**
- Return the key when iterating 
- Add support function for splitting keys

### CODE REVIEW

File: platform/view/services/kvs/kvs_test.go [hunk 1]
```diff
@@ -125,7 +125,7 @@ func testRound(t *testing.T, cfg *fakeProv) {
 
 	for ctr := 0; it.HasNext(); ctr++ {
 		val = &stuff{}
-		err = it.Next(val)
+		_, err = it.Next(val)
```
Comment (mbrandenburger) [inline platform/view/services/kvs/kvs_test.go:128]: can you please add a test case to check that we get the expected key as well?!

**Review by mbrandenburger — CHANGES_REQUESTED** [review]

File: platform/view/services/kvs/kvs_test.go (hunk 1 above)
Comment (adecaro) [reply platform/view/services/kvs/kvs_test.go]: done

File: platform/view/services/kvs/kvs_test.go [hunk 2]
```diff
@@ -125,9 +127,10 @@ func testRound(t *testing.T, cfg *fakeProv) {
...
 	for ctr := 0; it.HasNext(); ctr++ {
 		val = &stuff{}
-		_, err = it.Next(val)
+		key, err := it.Next(val)
 		assert.NoError(t, err)
 		if ctr == 0 {
+			assert.Equal(t, k1, key)
 			assert.Equal(t, &stuff{"santa", 1}, val)
 		} else {
 			assert.Fail(t, "expected 2 entries in the range, found more")
```
Comment (mbrandenburger) [inline platform/view/services/kvs/kvs_test.go:136]: do you want to take the chance to fix the error message here? It should be "expected 1 entry in the range, ..."

**Review by mbrandenburger — APPROVED** [review]
Thanks! LGTM

**Pull request merged**